---
title: "Updated results"
output: html_document
date: "2024-10-16"
---
```{r}
# Clean environment
rm( list = ls() )

# load libraries
library(xlsx)
library(tidyverse)
library(khroma)
library(colorBlindness)
library(ggplot2)

# load data
setwd("~/GitHub/BCH-GBS/2.Statistical_Analysis")
data <- read.xlsx("Clinical_Genomic_data.xlsx", sheetIndex = 1) 
# Remove contaminated samples
data_clean <- data|>
  filter(! ID %in% c("1_S1","32_S32","35_S35","81_S81"))|> 
  mutate(ICU = ifelse(Admission_loc=="ICU", 1,0))

```

# Invasive GBS surveillance
```{r}
# Number of patients 
nrow(data_clean)

# Isolation from blood vs CSF
table(data_clean$isolated_from)

# Age group at Diagnosis
table(data_clean$Age_cat)
table(data_clean$infant)

```
# Figure 1 - Table 
```{r}
# distribution of Serotypes Overall
Sero_df <- data_clean |> 
        group_by(Serotype)|> 
        summarise(N_p= n())|> 
        mutate("Total"= paste0(N_p, " (",round(100*(N_p/sum(N_p)),1), "%)"))|>
  select(-N_p)
# distribution of Serotypes in infants
Sero_df_infant <- data_clean |> 
        filter(infant ==1) |> 
        group_by(Serotype)|> 
        summarise(N_p= n())|> 
        mutate("Total Infants"= paste0(N_p, " (",round(100*(N_p/sum(N_p)),1), "%)"))|>
  select(-N_p)

# Age distribution of Serotypes
Age_Sero_df <- data_clean |>
  mutate(Age_cat = factor(Age_cat, levels = c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"))) |>
  group_by(Serotype, Age_cat) |>
  summarise(N = n(), .groups = "drop") |>
  pivot_wider(names_from = Age_cat, values_from = N) |>
  select(Serotype, "EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults")|>
  replace_na(list("EOD" = 0, "LOD" = 0, "VLOD" = 0, "Older Children (1-18 years)" = 0, "Adults" = 0))|>
  mutate(across(c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"), ~ paste(., " (", round(. / sum(.) * 100), "%)", sep="")))
  
# Join everything
Age_Sero_df_FINAL <- left_join(left_join(Age_Sero_df, Sero_df_infant), Sero_df) 

# distribution of CCs Overall
CC_df <- data_clean |> 
        group_by(CC)|> 
        summarise(N_p= n())|> 
        mutate("Total"= paste0(N_p, " (",round(100*(N_p/sum(N_p)),1), "%)"))|>
  select(-N_p)
# distribution of CCs in infants
CC_df_infant <- data_clean |> 
        filter(infant ==1) |> 
        group_by(CC)|> 
        summarise(N_p= n())|> 
        mutate("Total Infants"= paste0(N_p, " (",round(100*(N_p/sum(N_p)),1), "%)"))|>
  select(-N_p)

# Age distribution of CCs
Age_CC_df <- data_clean |>
  mutate(Age_cat = factor(Age_cat, levels = c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"))) |>
  group_by(CC, Age_cat) |>
  summarise(N = n(), .groups = "drop") |>
  pivot_wider(names_from = Age_cat, values_from = N) |>
  select(CC, "EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults")|>
  replace_na(list("EOD" = 0, "LOD" = 0, "VLOD" = 0, "Older Children (1-18 years)" = 0, "Adults" = 0))|>
  mutate(across(c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"), ~ paste(., " (", round(. / sum(.) * 100), "%)", sep="")))

# Join everything
Age_CC_df_FINAL <- left_join(left_join(Age_CC_df, CC_df_infant), CC_df) 


# Join for final table 
Age_Sero_CC_df <- rbind(Age_Sero_df_FINAL |> rename("group"="Serotype"), Age_CC_df_FINAL|> rename("group"="CC"))%>%
  mutate(group = factor(group, levels = c("Ia", "Ib", "II", "III", "IV", "V", "1", "12", "17", "19", "23", "459"))) %>%
  arrange(group)
Age_Sero_CC_df
```
# Figure 1 - Figure
```{r}
# Time distribution of CC
data_clean |> 
  mutate(year.of.cx = year(Date.of.cx))|> 
  group_by(CC, year.of.cx) |>
  summarise(Count = n(), .groups = 'drop') |>
  mutate(Proportion = Count / sum(Count)) |>
  ggplot(aes(x=as.factor(year.of.cx), y = Proportion, fill = as.factor(CC))) +
  geom_bar(stat = "identity") +
  theme_classic() +
  xlab("Year of Isolation") +
  ylab("Proportion") +
  labs(fill = "CC")+
  scale_fill_manual(values = color("light")(7))+
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
# Time distribution of Serotypes
data_clean |> 
  mutate(year.of.cx = year(Date.of.cx))|> 
  group_by(Serotype, year.of.cx) |>
  summarise(Count = n(), .groups = 'drop') |>
  mutate(Proportion = Count / sum(Count)) |>
  ggplot(aes(x=as.factor(year.of.cx), y = Proportion, fill = Serotype)) +
  geom_bar(stat = "identity") +
  theme_classic() +
  xlab("Year of Isolation") +
  ylab("Proportion") +
  labs(fill = "Serotype")+
  scale_fill_manual(values = color("vibrant")(6))+
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
```




# Clinical Characteristics
## Table 1
```{r}
# Difference in laboratory parameters between infants and older children on admission.

## P-values
p_val_Hb <- round(wilcox.test(Hb ~ infant, data = data_clean)$p.value, 3)
p_val_WBC <- round(wilcox.test(WBC ~ infant, data = data_clean)$p.value, 3)
p_val_Leukocytosis <- round(coef(summary(glm(Leukocytosis ~ infant, data_clean, family = binomial)))[2,4], 3)
p_val_Leukopenia <- round(coef(summary(glm(Leukopenia ~ infant, data_clean, family = binomial)))[2,4], 3)
p_val_platelet <- round(wilcox.test(platelet ~ infant, data = data_clean)$p.value, 3)
p_val_ANC <- round(wilcox.test(ANC ~ infant, data = data_clean)$p.value, 3)
p_val_Neutropenia <- round(coef(summary(glm(Neutropenia ~ infant, data_clean, family = binomial)))[2,4], 3)
## Table
Table1_1 <- data_clean |> 
  group_by(infant)|>
  summarise(N=n(), 
            Hb_mean_SD= paste(round(mean(Hb, na.rm=T),2), " (", round(sd(Hb, na.rm=T), 2),")", sep=""), 
            WBC_mean_SD= paste(round(mean(WBC, na.rm=T),2), " (", round(sd(WBC, na.rm=T),2),")", sep=""), 
            Leukocytosis_Yes_p = paste(sum(Leukocytosis==1, na.rm = T), " (", round(100*mean(Leukocytosis==1, na.rm = T), 1),")", sep=""), 
            Leukopenia_Yes_p  = paste(sum(Leukopenia==1, na.rm = T), " (", round(100*mean(Leukopenia==1, na.rm = T), 1),")", sep=""), 
            Platelet_mean_SD= paste(round(mean(platelet, na.rm=T),2), " (",  round(sd(platelet, na.rm=T),2),")", sep=""), 
            ANC_mean_SD= paste(round(mean(ANC, na.rm=T),2), " (",  round(sd(ANC, na.rm=T),2),")", sep=""), 
            Neutropenia_Yes_p = paste(sum(Neutropenia==1, na.rm = T), " (",  round(100*mean(Neutropenia==1, na.rm = T),1),")", sep="")) |>
  mutate(Value= paste0(infant, "(N=", N, ")"), .before =Hb_mean_SD)|>
  select(-infant, -N)
Table1_1 <- rbind(Table1_1, c("p-value", p_val_Hb, p_val_WBC, p_val_Leukocytosis, p_val_Leukopenia, p_val_platelet, p_val_ANC,p_val_Neutropenia))

# ICU vs other  
## P-values
p_val_2_Hb <- round(coef(summary(glm(ICU ~ Hb, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
p_val_2_WBC <- round(coef(summary(glm(ICU ~ WBC, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
p_val_2_Leukocytosis <- round(coef(summary(glm(ICU ~ Leukocytosis, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
p_val_2_Leukopenia <- round(coef(summary(glm(ICU ~ Leukopenia, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
p_val_2_platelet <- round(coef(summary(glm(ICU ~ platelet, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
p_val_2_ANC <- round(coef(summary(glm(ICU ~ ANC, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
p_val_2_Neutropenia <- round(coef(summary(glm(ICU ~ Neutropenia, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
## Table
Table1_2 <- data_clean |> 
  mutate(Admission_loc= factor(Admission_loc, levels= c("other", "ICU")))|>
  filter(days_at_dx <= 365)|>
  group_by(Admission_loc)|>
  summarise(N=n(), 
            Hb_mean_SD= paste(round(mean(Hb, na.rm=T),2), " (", round(sd(Hb, na.rm=T), 2),")", sep=""), 
            WBC_mean_SD= paste(round(mean(WBC, na.rm=T),2), " (", round(sd(WBC, na.rm=T),2),")", sep=""), 
            Leukocytosis_Yes_p = paste(sum(Leukocytosis==1, na.rm = T), " (", round(100*mean(Leukocytosis==1, na.rm = T), 1),")", sep=""), 
            Leukopenia_Yes_p  = paste(sum(Leukopenia==1, na.rm = T), " (", round(100*mean(Leukopenia==1, na.rm = T), 1),")", sep=""), 
            Platelet_mean_SD= paste(round(mean(platelet, na.rm=T),2), " (",  round(sd(platelet, na.rm=T),2),")", sep=""), 
            ANC_mean_SD= paste(round(mean(ANC, na.rm=T),2), " (",  round(sd(ANC, na.rm=T),2),")", sep=""), 
            Neutropenia_Yes_p = paste(sum(Neutropenia==1, na.rm = T), " (",  round(100*mean(Neutropenia==1, na.rm = T),1),")", sep="")) |>
  mutate(Value= paste0(Admission_loc, "(N=", N, ")"), .before =Hb_mean_SD)|>
  select(-Admission_loc, -N)
Table1_2 <- rbind(Table1_2, c("p-value", p_val_2_Hb, p_val_2_WBC, p_val_2_Leukocytosis, p_val_2_Leukopenia, p_val_2_platelet, p_val_2_ANC,p_val_2_Neutropenia))


# ICU vs other  
## P-values
p_val_3_Hb <- round(wilcox.test(Hb ~ Meningitis, data = data_clean|> filter(days_at_dx <= 365))$p.value, 3)
p_val_3_WBC <- round(wilcox.test(WBC ~ Meningitis, data = data_clean|> filter(days_at_dx <= 365))$p.value, 3)
p_val_3_Leukocytosis <- round(coef(summary(glm(Leukocytosis ~ Meningitis, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
p_val_3_Leukopenia <- round(coef(summary(glm(Leukopenia ~ Meningitis, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
p_val_3_platelet <- round(wilcox.test(platelet ~ Meningitis, data = data_clean|> filter(days_at_dx <= 365))$p.value, 3)
p_val_3_ANC <- round(wilcox.test(ANC ~ Meningitis, data = data_clean|> filter(days_at_dx <= 365))$p.value, 3)
p_val_3_Neutropenia <- round(coef(summary(glm(Neutropenia ~ Meningitis, data_clean|> filter(days_at_dx <= 365), family = binomial)))[2,4], 3)
## Table
Table1_3 <- data_clean |> 
  filter(days_at_dx <= 365)|>
  group_by(Meningitis)|>
  summarise(N=n(), 
            Hb_mean_SD= paste(round(mean(Hb, na.rm=T),2), " (", round(sd(Hb, na.rm=T), 2),")", sep=""), 
            WBC_mean_SD= paste(round(mean(WBC, na.rm=T),2), " (", round(sd(WBC, na.rm=T),2),")", sep=""), 
            Leukocytosis_Yes_p = paste(sum(Leukocytosis==1, na.rm = T), " (", round(100*mean(Leukocytosis==1, na.rm = T), 1),")", sep=""), 
            Leukopenia_Yes_p  = paste(sum(Leukopenia==1, na.rm = T), " (", round(100*mean(Leukopenia==1, na.rm = T), 1),")", sep=""), 
            Platelet_mean_SD= paste(round(mean(platelet, na.rm=T),2), " (",  round(sd(platelet, na.rm=T),2),")", sep=""), 
            ANC_mean_SD= paste(round(mean(ANC, na.rm=T),2), " (",  round(sd(ANC, na.rm=T),2),")", sep=""), 
            Neutropenia_Yes_p = paste(sum(Neutropenia==1, na.rm = T), " (",  round(100*mean(Neutropenia==1, na.rm = T),1),")", sep="")) |>
  mutate(Value= paste0(Meningitis, "(N=", N, ")"), .before =Hb_mean_SD)|>
  select(-Meningitis, -N)
Table1_3 <- rbind(Table1_3, c("p-value", p_val_3_Hb, p_val_3_WBC, p_val_3_Leukocytosis, p_val_3_Leukopenia, p_val_3_platelet, p_val_3_ANC,p_val_3_Neutropenia))
  
rbind(Table1_1, Table1_2, Table1_3)
```


# Molecular characteristics of the bacterial population
```{r}
# CC distribution
data_clean |>
  group_by(CC)|>
  summarise(N= n())|>
  mutate(p= round(100*(N/sum(N))))|>
  arrange(-N)

# ST distribution
data_clean |>
  group_by(ST)|>
  summarise(N= n())|>
  mutate(p= round(100*(N/sum(N))))|>
  arrange(-N)

# STs by CC distribution
data_clean |>
  group_by(CC, ST)|>
  summarise(N= n())|>
  mutate(p= round(100*(N/sum(N))))

# Serotype distribution of CCs
data_clean |>
  mutate(CC= paste("CC", CC, sep=""))|>
  group_by(Serotype, CC) |>
  summarise(N = n(), .groups = "drop") |>
  pivot_wider(names_from = CC, values_from = N) |>
  select(Serotype, "CC1", "CC12", "CC17","CC19", "CC23", "CC459","CCNA")|>
  rename("NF"="CCNA")|>
  replace_na(list("CC1" = 0, "CC12" = 0, "CC17" = 0, "CC19" = 0, "CC23" = 0, "CC459" = 0, "NF" = 0))|>
  mutate(across(c("CC1", "CC12", "CC17","CC19", "CC23", "CC459", "NF"), ~ paste(., " (", round(. / sum(.) * 100), "%)", sep="")))
```

## Supp. Table 3. 
```{r}
# Age distribution of CCs
CC_Age_df <- data_clean |>
  mutate(Age_cat = factor(Age_cat, levels = c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"))) |>
  group_by(CC, Age_cat) |>
  summarise(N = n(), .groups = "drop") |>
  pivot_wider(names_from = Age_cat, values_from = N) |>
  select(CC, "EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults")|>
  replace_na(list("EOD" = 0, "LOD" = 0, "VLOD" = 0, "Older Children (1-18 years)" = 0, "Adults" = 0))|>
  mutate(across(c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"), ~ paste(., " (", round(. / sum(.) * 100), "%)", sep="")))|>
  mutate(ST = "Total", .before = "EOD")|>
  filter(!is.na(CC ))

# Age distribution of STs
ST_Age_df <- data_clean |>
  mutate(Age_cat = factor(Age_cat, levels = c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"))) |>
  group_by(CC, ST, Age_cat) |>
  summarise(N = n(), .groups = "drop") |>
  pivot_wider(names_from = Age_cat, values_from = N) |>
  select(CC, ST, "EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults")|>
  replace_na(list("EOD" = 0, "LOD" = 0, "VLOD" = 0, "Older Children (1-18 years)" = 0, "Adults" = 0))|>
  mutate(across(c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"), ~ paste(., " (", round(. / sum(.) * 100), "%)", sep="")))|>
  mutate(ST= paste0("ST",ST))
# Join
CC_ST_Age_df <- ST_Age_df
for(i in 1:nrow(CC_Age_df)){
  if(nrow(ST_Age_df |> filter(CC == CC_Age_df$CC[i])) >1 ){
    rows <- which(CC_ST_Age_df$CC == CC_Age_df$CC[i])
    CC_ST_Age_df <- CC_ST_Age_df |> add_row(CC_Age_df[i,], .before = last(rows)+1)
  }
}

# Total columns
CC_ST_Total_df <- CC_ST_Age_df |>
  mutate(across(c("EOD", "LOD", "VLOD", "Older Children (1-18 years)", "Adults"),
                ~ as.numeric(str_split(., "\\(", simplify = TRUE)[,1]))) |>  # Convert extracted values to numeric
  rowwise() |>  # Apply the sum row-wise
  mutate(Total = sum(c_across(c("EOD", "LOD", "VLOD", "Older Children (1-18 years)", "Adults")), na.rm = TRUE)) |>  # Sum across the selected columns
  ungroup()|>
  select(CC, ST, Total)
  

# Final Table 
CC_ST_Age_df_FINAL <- left_join(CC_ST_Age_df, CC_ST_Total_df)
CC_ST_Age_df_FINAL
```
## Supp. Table 4. 
```{r}
# ST distribution
ST_df_0 <- data_clean |>
  group_by(ST)|>
  summarise(N= n())|>
  mutate(p= paste0("(",round(100*(N/sum(N))), "%)"))|>
  arrange(-N)

# Know ST profiles
ST_df_1 <- unique(data_clean |>
  select(contains("ST"))|>
  filter(ST_info.mismatches == "0" & ST != "NF")|>
  select(-ST_info.uncertainty, -ST_info.depth, -ST_info.maxMAF) )

# Known ST profiles with mismatches
ST_df_2 <- data_clean |>
  select(contains("ST"))|>
  filter(ST_info.mismatches != "0")|>
  select(-ST_info.uncertainty, -ST_info.depth, -ST_info.maxMAF) 

# New ST profile
ST_df_3 <- data_clean |>
  select(contains("ST"))|>
  filter(ST== "NF")|>
  select(-ST_info.uncertainty, -ST_info.depth, -ST_info.maxMAF) 

# Combine
ST_df <- left_join(ST_df_0, rbind(ST_df_1, ST_df_2, ST_df_3)) 
colnames(ST_df) <- str_remove(colnames(ST_df), "ST_info.")
ST_df
```


# Coverage of genes encoding proteins of vaccine targets 
## distribution of Surface protein and virulence factors across all Age groups
```{r}
Surf_prot_df <- data_clean |> 
  mutate(Surface_protein.ALP_FAM = Surface_protein.ALPHA + Surface_protein.ALP1 + Surface_protein.ALP23 + Surface_protein.RIB,
         Surface_protein.PILUS_island = rowSums(across(contains("PI"))), 
         Surface_protein.SRR = Surface_protein.SRR1+Surface_protein.SRR2)|>
  summarise(N=n(),
            Trivalent = paste(sum(Serotype %in% c("Ia", "Ib", "III")), " (",round(100*mean(Serotype %in% c("Ia", "Ib", "III"))), "%)", sep=""), 
            Hexavalent = paste(sum(Serotype %in% c("Ia", "Ib", "II", "III", "IV", "V")), " (",round(100*mean(Serotype %in% c("Ia", "Ib", "II", "III", "IV", "V"))), "%)", sep=""),
            Surface_protein.RIB = paste(sum(Surface_protein.RIB ==1), " (",round(100*mean(Surface_protein.RIB ==1)), "%)", sep=""), 
            Surface_protein.ALPHA = paste(sum(Surface_protein.ALPHA ==1), " (",round(100*mean(Surface_protein.ALPHA ==1)), "%)", sep=""), 
            Surface_protein.ALP1 = paste(sum(Surface_protein.ALP1 ==1), " (",round(100*mean(Surface_protein.ALP1 ==1)), "%)", sep=""), 
            Surface_protein.ALP23 = paste(sum(Surface_protein.ALP23 ==1), " (",round(100*mean(Surface_protein.ALP23 ==1)), "%)", sep=""), 
            Surface_protein.ALP_FAM= paste(sum(Surface_protein.ALP_FAM>0), " (",round(100*mean(Surface_protein.ALP_FAM>0)), "%)", sep=""), 
            Surface_protein.PI1  = paste(sum(Surface_protein.PI1  ==1), " (",round(100*mean(Surface_protein.PI1  ==1)), "%)", sep=""), 
            Surface_protein.PI2A1  = paste(sum(Surface_protein.PI2A1 ==1), " (",round(100*mean(Surface_protein.PI2A1  ==1)), "%)", sep=""), 
            Surface_protein.PI2A2  = paste(sum(Surface_protein.PI2A2 ==1), " (",round(100*mean(Surface_protein.PI2A2  ==1)), "%)", sep=""), 
            Surface_protein.PI2B = paste(sum(Surface_protein.PI2B ==1), " (",round(100*mean(Surface_protein.PI2B ==1)), "%)", sep=""), 
            Surface_protein.PILUS_island= paste(sum(Surface_protein.PILUS_island>0), " (",round(100*mean(Surface_protein.PILUS_island>0)), "%)", sep=""), 
           Surface_protein.SRR1 = paste(sum(Surface_protein.SRR1 ==1), " (",round(100*mean(Surface_protein.SRR1 ==1)), "%)", sep=""), 
           Surface_protein.SRR2 = paste(sum(Surface_protein.SRR2 ==1), " (",round(100*mean(Surface_protein.SRR2 ==1)), "%)", sep=""), 
           Surface_protein.SRR = paste(sum(Surface_protein.SRR ==1), " (",round(100*mean(Surface_protein.SRR ==1)), "%)", sep=""), 
           Surface_protein.Sip.1a = paste(sum(Surface_protein.Sip.1a ==1), " (",round(100*mean(Surface_protein.Sip.1a ==1)), "%)", sep=""), 
           Surface_protein.Sip.3a = paste(sum(Surface_protein.Sip.3a ==1), " (",round(100*mean(Surface_protein.Sip.3a ==1)), "%)", sep=""), 
           Surface_protein.Sip = paste(sum(Surface_protein.Sip ==1), " (",round(100*mean(Surface_protein.Sip ==1)), "%)", sep=""), 
           Surface_protein.HVGA = paste(sum(Surface_protein.HVGA ==1), " (",round(100*mean(Surface_protein.HVGA ==1)), "%)", sep=""), 
           Virulence_Gene.scpB = paste(sum(Virulence_Gene.scpB ==1), " (",round(100*mean(Virulence_Gene.scpB ==1)), "%)", sep=""), 
           Virulence_Gene.lmb = paste(sum(Virulence_Gene.lmb ==1), " (",round(100*mean(Virulence_Gene.lmb ==1)), "%)", sep=""), 
           Virulence_Gene.hylB = paste(sum(Virulence_Gene.hylB ==1), " (",round(100*mean(Virulence_Gene.hylB ==1)), "%)", sep=""), 
           Virulence_Gene.fbsB = paste(sum(Virulence_Gene.fbsB ==1), " (",round(100*mean(Virulence_Gene.fbsB ==1)), "%)", sep=""))
Surf_prot_df
```
##  distribution of Surface protein and virulence factors across all INFANTS
```{r}
Surf_prot_Infant_df <- data_clean |> 
  filter(days_at_dx <= 365) |> 
  mutate(Surface_protein.ALP_FAM = Surface_protein.ALPHA + Surface_protein.ALP1 + Surface_protein.ALP23 + Surface_protein.RIB,
         Surface_protein.PILUS_island = rowSums(across(contains("PI"))), 
         Surface_protein.SRR = Surface_protein.SRR1+Surface_protein.SRR2)|>
  summarise(N=n(),
            Trivalent = paste(sum(Serotype %in% c("Ia", "Ib", "III")), " (",round(100*mean(Serotype %in% c("Ia", "Ib", "III"))), "%)", sep=""), 
            Hexavalent = paste(sum(Serotype %in% c("Ia", "Ib", "II", "III", "IV", "V")), " (",round(100*mean(Serotype %in% c("Ia", "Ib", "II", "III", "IV", "V"))), "%)", sep=""),
            Surface_protein.RIB = paste(sum(Surface_protein.RIB ==1), " (",round(100*mean(Surface_protein.RIB ==1)), "%)", sep=""), 
            Surface_protein.ALPHA = paste(sum(Surface_protein.ALPHA ==1), " (",round(100*mean(Surface_protein.ALPHA ==1)), "%)", sep=""), 
            Surface_protein.ALP1 = paste(sum(Surface_protein.ALP1 ==1), " (",round(100*mean(Surface_protein.ALP1 ==1)), "%)", sep=""), 
            Surface_protein.ALP23 = paste(sum(Surface_protein.ALP23 ==1), " (",round(100*mean(Surface_protein.ALP23 ==1)), "%)", sep=""), 
            Surface_protein.ALP_FAM= paste(sum(Surface_protein.ALP_FAM>0), " (",round(100*mean(Surface_protein.ALP_FAM>0)), "%)", sep=""), 
            Surface_protein.PI1  = paste(sum(Surface_protein.PI1  ==1), " (",round(100*mean(Surface_protein.PI1  ==1)), "%)", sep=""), 
            Surface_protein.PI2A1  = paste(sum(Surface_protein.PI2A1 ==1), " (",round(100*mean(Surface_protein.PI2A1  ==1)), "%)", sep=""), 
            Surface_protein.PI2A2  = paste(sum(Surface_protein.PI2A2 ==1), " (",round(100*mean(Surface_protein.PI2A2  ==1)), "%)", sep=""), 
            Surface_protein.PI2B = paste(sum(Surface_protein.PI2B ==1), " (",round(100*mean(Surface_protein.PI2B ==1)), "%)", sep=""), 
            Surface_protein.PILUS_island= paste(sum(Surface_protein.PILUS_island>0), " (",round(100*mean(Surface_protein.PILUS_island>0)), "%)", sep=""), 
            Surface_protein.HVGA = paste(sum(Surface_protein.HVGA ==1), " (",round(100*mean(Surface_protein.HVGA ==1)), "%)", sep=""), 
           Surface_protein.SRR1 = paste(sum(Surface_protein.SRR1 ==1), " (",round(100*mean(Surface_protein.SRR1 ==1)), "%)", sep=""), 
           Surface_protein.SRR2 = paste(sum(Surface_protein.SRR2 ==1), " (",round(100*mean(Surface_protein.SRR2 ==1)), "%)", sep=""), 
           Surface_protein.SRR = paste(sum(Surface_protein.SRR ==1), " (",round(100*mean(Surface_protein.SRR ==1)), "%)", sep=""), 
           Surface_protein.Sip.1a = paste(sum(Surface_protein.Sip.1a ==1), " (",round(100*mean(Surface_protein.Sip.1a ==1)), "%)", sep=""), 
           Surface_protein.Sip.3a = paste(sum(Surface_protein.Sip.3a ==1), " (",round(100*mean(Surface_protein.Sip.3a ==1)), "%)", sep=""), 
           Surface_protein.Sip = paste(sum(Surface_protein.Sip ==1), " (",round(100*mean(Surface_protein.Sip ==1)), "%)", sep=""), 
           Virulence_Gene.scpB = paste(sum(Virulence_Gene.scpB ==1), " (",round(100*mean(Virulence_Gene.scpB ==1)), "%)", sep=""), 
           Virulence_Gene.lmb = paste(sum(Virulence_Gene.lmb ==1), " (",round(100*mean(Virulence_Gene.lmb ==1)), "%)", sep=""), 
           Virulence_Gene.hylB = paste(sum(Virulence_Gene.hylB ==1), " (",round(100*mean(Virulence_Gene.hylB ==1)), "%)", sep=""), 
           Virulence_Gene.fbsB = paste(sum(Virulence_Gene.fbsB ==1), " (",round(100*mean(Virulence_Gene.fbsB ==1)), "%)", sep=""))
Surf_prot_Infant_df
```
##  distribution of ALP family proteins per CC
```{r}
Surf_prot_per_CC_df <- data_clean |>
    filter(days_at_dx <= 365) |> 
  group_by(CC)|> 
  mutate(Surface_protein.ALP_FAM = Surface_protein.ALPHA + Surface_protein.ALP1 + Surface_protein.ALP23 + Surface_protein.RIB,
         Surface_protein.PILUS_island = rowSums(across(contains("PI"))), 
         Surface_protein.SRR = Surface_protein.SRR1+Surface_protein.SRR2)|>
  summarise(N=n(),
            Trivalent = paste(sum(Serotype %in% c("Ia", "Ib", "III")), " (",round(100*mean(Serotype %in% c("Ia", "Ib", "III"))), "%)", sep=""), 
            Hexavalent = paste(sum(Serotype %in% c("Ia", "Ib", "II", "III", "IV", "V")), " (",round(100*mean(Serotype %in% c("Ia", "Ib", "II", "III", "IV", "V"))), "%)", sep=""),
            Surface_protein.RIB = paste(sum(Surface_protein.RIB ==1), " (",round(100*mean(Surface_protein.RIB ==1)), "%)", sep=""), 
            Surface_protein.ALPHA = paste(sum(Surface_protein.ALPHA ==1), " (",round(100*mean(Surface_protein.ALPHA ==1)), "%)", sep=""), 
            Surface_protein.ALP1 = paste(sum(Surface_protein.ALP1 ==1), " (",round(100*mean(Surface_protein.ALP1 ==1)), "%)", sep=""), 
            Surface_protein.ALP23 = paste(sum(Surface_protein.ALP23 ==1), " (",round(100*mean(Surface_protein.ALP23 ==1)), "%)", sep=""), 
            Surface_protein.ALP_FAM= paste(sum(Surface_protein.ALP_FAM>0), " (",round(100*mean(Surface_protein.ALP_FAM>0)), "%)", sep=""), 
            Surface_protein.PI1  = paste(sum(Surface_protein.PI1  ==1), " (",round(100*mean(Surface_protein.PI1  ==1)), "%)", sep=""), 
            Surface_protein.PI2A1  = paste(sum(Surface_protein.PI2A1 ==1), " (",round(100*mean(Surface_protein.PI2A1  ==1)), "%)", sep=""), 
            Surface_protein.PI2A2  = paste(sum(Surface_protein.PI2A2 ==1), " (",round(100*mean(Surface_protein.PI2A2  ==1)), "%)", sep=""), 
            Surface_protein.PI2B = paste(sum(Surface_protein.PI2B ==1), " (",round(100*mean(Surface_protein.PI2B ==1)), "%)", sep=""), 
            Surface_protein.PILUS_island= paste(sum(Surface_protein.PILUS_island>0), " (",round(100*mean(Surface_protein.PILUS_island>0)), "%)", sep=""), 
            Surface_protein.HVGA = paste(sum(Surface_protein.HVGA ==1), " (",round(100*mean(Surface_protein.HVGA ==1)), "%)", sep=""), 
           Surface_protein.SRR1 = paste(sum(Surface_protein.SRR1 ==1), " (",round(100*mean(Surface_protein.SRR1 ==1)), "%)", sep=""), 
           Surface_protein.SRR2 = paste(sum(Surface_protein.SRR2 ==1), " (",round(100*mean(Surface_protein.SRR2 ==1)), "%)", sep=""), 
           Surface_protein.SRR = paste(sum(Surface_protein.SRR ==1), " (",round(100*mean(Surface_protein.SRR ==1)), "%)", sep=""), 
           Surface_protein.Sip.1a = paste(sum(Surface_protein.Sip.1a ==1), " (",round(100*mean(Surface_protein.Sip.1a ==1)), "%)", sep=""), 
           Surface_protein.Sip.3a = paste(sum(Surface_protein.Sip.3a ==1), " (",round(100*mean(Surface_protein.Sip.3a ==1)), "%)", sep=""), 
           Surface_protein.Sip = paste(sum(Surface_protein.Sip ==1), " (",round(100*mean(Surface_protein.Sip ==1)), "%)", sep=""), 
           Virulence_Gene.scpB = paste(sum(Virulence_Gene.scpB ==1), " (",round(100*mean(Virulence_Gene.scpB ==1)), "%)", sep=""), 
           Virulence_Gene.lmb = paste(sum(Virulence_Gene.lmb ==1), " (",round(100*mean(Virulence_Gene.lmb ==1)), "%)", sep=""), 
           Virulence_Gene.hylB = paste(sum(Virulence_Gene.hylB ==1), " (",round(100*mean(Virulence_Gene.hylB ==1)), "%)", sep=""), 
           Virulence_Gene.fbsB = paste(sum(Virulence_Gene.fbsB ==1), " (",round(100*mean(Virulence_Gene.fbsB ==1)), "%)", sep=""))|> 
  mutate(CC= paste("CC", CC, " (N=", N, ")", sep=""))|> 
  select(-N)
Surf_prot_per_CC_df_t <- as.data.frame(t(Surf_prot_per_CC_df))
colnames(Surf_prot_per_CC_df_t) <- Surf_prot_per_CC_df_t[1,]
Surf_prot_per_CC_df_t <- Surf_prot_per_CC_df_t[-1, ]
Surf_prot_per_CC_df_t
```


## Supp. Table 5
# distribution of ALP family proteins per Age group
```{r}
Surf_prot_per_Age_cat_df <- data_clean |>
  mutate(Age_cat = factor(Age_cat, levels = c("EOD", "LOD", "VLOD","Older Children (1-18 years)", "Adults"))) |>
  group_by(Age_cat)|> 
  mutate(Surface_protein.ALP_FAM = Surface_protein.ALPHA + Surface_protein.ALP1 + Surface_protein.ALP23 + Surface_protein.RIB,
         Surface_protein.PILUS_island = rowSums(across(contains("PI"))), 
         Surface_protein.SRR = Surface_protein.SRR1+Surface_protein.SRR2)|>
  summarise(N=n(),
            Trivalent = paste(sum(Serotype %in% c("Ia", "Ib", "III")), " (",round(100*mean(Serotype %in% c("Ia", "Ib", "III"))), "%)", sep=""), 
            Hexavalent = paste(sum(Serotype %in% c("Ia", "Ib", "II", "III", "IV", "V")), " (",round(100*mean(Serotype %in% c("Ia", "Ib", "II", "III", "IV", "V"))), "%)", sep=""),
            Surface_protein.RIB = paste(sum(Surface_protein.RIB ==1), " (",round(100*mean(Surface_protein.RIB ==1)), "%)", sep=""), 
            Surface_protein.ALPHA = paste(sum(Surface_protein.ALPHA ==1), " (",round(100*mean(Surface_protein.ALPHA ==1)), "%)", sep=""), 
            Surface_protein.ALP1 = paste(sum(Surface_protein.ALP1 ==1), " (",round(100*mean(Surface_protein.ALP1 ==1)), "%)", sep=""), 
            Surface_protein.ALP23 = paste(sum(Surface_protein.ALP23 ==1), " (",round(100*mean(Surface_protein.ALP23 ==1)), "%)", sep=""), 
            Surface_protein.ALP_FAM= paste(sum(Surface_protein.ALP_FAM>0), " (",round(100*mean(Surface_protein.ALP_FAM>0)), "%)", sep=""), 
            Surface_protein.PI1  = paste(sum(Surface_protein.PI1  ==1), " (",round(100*mean(Surface_protein.PI1  ==1)), "%)", sep=""), 
            Surface_protein.PI2A1  = paste(sum(Surface_protein.PI2A1 ==1), " (",round(100*mean(Surface_protein.PI2A1  ==1)), "%)", sep=""), 
            Surface_protein.PI2A2  = paste(sum(Surface_protein.PI2A2 ==1), " (",round(100*mean(Surface_protein.PI2A2  ==1)), "%)", sep=""), 
            Surface_protein.PI2B = paste(sum(Surface_protein.PI2B ==1), " (",round(100*mean(Surface_protein.PI2B ==1)), "%)", sep=""), 
            Surface_protein.PILUS_island= paste(sum(Surface_protein.PILUS_island>0), " (",round(100*mean(Surface_protein.PILUS_island>0)), "%)", sep=""), 
            Surface_protein.HVGA = paste(sum(Surface_protein.HVGA ==1), " (",round(100*mean(Surface_protein.HVGA ==1)), "%)", sep=""), 
           Surface_protein.SRR1 = paste(sum(Surface_protein.SRR1 ==1), " (",round(100*mean(Surface_protein.SRR1 ==1)), "%)", sep=""), 
           Surface_protein.SRR2 = paste(sum(Surface_protein.SRR2 ==1), " (",round(100*mean(Surface_protein.SRR2 ==1)), "%)", sep=""), 
           Surface_protein.SRR = paste(sum(Surface_protein.SRR ==1), " (",round(100*mean(Surface_protein.SRR ==1)), "%)", sep=""), 
           Surface_protein.Sip.1a = paste(sum(Surface_protein.Sip.1a ==1), " (",round(100*mean(Surface_protein.Sip.1a ==1)), "%)", sep=""), 
           Surface_protein.Sip.3a = paste(sum(Surface_protein.Sip.3a ==1), " (",round(100*mean(Surface_protein.Sip.3a ==1)), "%)", sep=""), 
           Surface_protein.Sip = paste(sum(Surface_protein.Sip ==1), " (",round(100*mean(Surface_protein.Sip ==1)), "%)", sep=""), 
           Virulence_Gene.scpB = paste(sum(Virulence_Gene.scpB ==1), " (",round(100*mean(Virulence_Gene.scpB ==1)), "%)", sep=""), 
           Virulence_Gene.lmb = paste(sum(Virulence_Gene.lmb ==1), " (",round(100*mean(Virulence_Gene.lmb ==1)), "%)", sep=""), 
           Virulence_Gene.hylB = paste(sum(Virulence_Gene.hylB ==1), " (",round(100*mean(Virulence_Gene.hylB ==1)), "%)", sep=""), 
           Virulence_Gene.fbsB = paste(sum(Virulence_Gene.fbsB ==1), " (",round(100*mean(Virulence_Gene.fbsB ==1)), "%)", sep=""))|> 
  mutate(Age_cat= paste(Age_cat, " (N=", N, ")", sep=""))|> 
  select(-N)
Surf_prot_per_Age_cat_df_t <- cbind(as.data.frame(t(Surf_prot_per_Age_cat_df)), as.data.frame(t(Surf_prot_Infant_df)), as.data.frame(t(Surf_prot_df)))
colnames(Surf_prot_per_Age_cat_df_t) <- Surf_prot_per_Age_cat_df_t[1,]
colnames(Surf_prot_per_Age_cat_df_t)[length(colnames(Surf_prot_per_Age_cat_df_t))-1] <- "Total Infants (N=70)"
colnames(Surf_prot_per_Age_cat_df_t)[length(colnames(Surf_prot_per_Age_cat_df_t))] <- "Total (N=87)"
Surf_prot_per_Age_cat_df_t <- Surf_prot_per_Age_cat_df_t[-1, ]
Surf_prot_per_Age_cat_df_t
```



# Overlap between phenotypic antibiotic resistance and presence of known genotypic markers
## Phenotypic resistance testing 
```{r}
s_df<- data_clean |>
  summarise(Penicillin= paste(sum(Penicillin =="s"), " (",round(100*mean(Penicillin =="s")), "%)", sep=""), 
            Vancomycin= paste(sum(Vancomycin =="s"), " (",round(100*mean(Vancomycin =="s")), "%)", sep=""), 
            Erythromycin= paste(sum(Erythromycin =="s"), " (",round(100*mean(Erythromycin =="s")), "%)", sep=""), 
            Clindamycin= paste(sum(Clindamycin =="s"), " (",round(100*mean(Clindamycin =="s")), "%)", sep=""))
i_df<- data_clean |>
  summarise(Penicillin= paste(sum(Penicillin =="i"), " (",round(100*mean(Penicillin =="i")), "%)", sep=""), 
            Vancomycin= paste(sum(Vancomycin =="i"), " (",round(100*mean(Vancomycin =="i")), "%)", sep=""), 
            Erythromycin= paste(sum(Erythromycin =="i"), " (",round(100*mean(Erythromycin =="i")), "%)", sep=""), 
            Clindamycin= paste(sum(Clindamycin =="i"), " (",round(100*mean(Clindamycin =="i")), "%)", sep=""))
r_df<- data_clean |>
  summarise(Penicillin= paste(sum(Penicillin =="r"), " (",round(100*mean(Penicillin =="r")), "%)", sep=""), 
            Vancomycin= paste(sum(Vancomycin =="r"), " (",round(100*mean(Vancomycin =="r")), "%)", sep=""), 
            Erythromycin= paste(sum(Erythromycin =="r"), " (",round(100*mean(Erythromycin =="r")), "%)", sep=""), 
            Clindamycin= paste(sum(Clindamycin =="r"), " (",round(100*mean(Clindamycin =="r")), "%)", sep=""))
pheno_res_df<- rbind(s_df, i_df, r_df)|>
  mutate(Resistance = c("Susceptible", "Intermediate Resistance", "Resistant"))|>
  select(Resistance, Penicillin, Vancomycin, Erythromycin, Clindamycin)
pheno_res_df


```
## In-silico detection resistance genes
### Supplementary Tab 2 (5 in the main text)
```{r}
### Overall
In_silico_df <- data_clean |>
  mutate(Res_MUT.X23S1_SNP= ifelse(Res_MUT.X23S1_SNP != "0", 1, 0),
         Res_MUT.X23S3_SNP= ifelse(Res_MUT.X23S3_SNP != "0", 1, 0),
         Res_MUT.GYRA_SNP= ifelse(Res_MUT.GYRA_SNP != "0", 1, 0),
         Res_MUT.PARC_SNP= ifelse(Res_MUT.PARC_SNP != "0", 1, 0))|>
  summarise(N=n(),
            ANT6IA=paste(sum(Res_Gene.ANT6IA), " (", round(100*mean(Res_Gene.ANT6IA==1)), "%)",sep=""),
            APH3III=paste(sum(Res_Gene.APH3III)," (", round(100*mean(Res_Gene.APH3III==1)), "%)",sep=""),,
            Aminoglycosides=paste(sum(Res_Gene.ANT6IA+Res_Gene.APH3III!=0), " (", round(100*mean(Res_Gene.ANT6IA+Res_Gene.APH3III!=0)), "%)",sep=""),
            TETM=paste(sum(Res_Gene.TETM)," (", round(100*mean(Res_Gene.TETM==1)), "%)",sep=""),
            TETO=paste(sum(Res_Gene.TETO)," (", round(100*mean(Res_Gene.TETO==1)), "%)",sep=""),
            Tetracyclines=paste(sum(Res_Gene.TETM+Res_Gene.TETO !=0), " (", round(100*mean(Res_Gene.TETM+Res_Gene.TETO !=0)), "%)",sep=""),
            ERMA=paste(sum(Res_Gene.ERMA)," (", round(100*mean(Res_Gene.ERMA==1)), "%)",sep=""),
            ERMB=paste(sum(Res_Gene.ERMB)," (", round(100*mean(Res_Gene.ERMB==1)), "%)",sep=""),
            ERMT=paste(sum(Res_Gene.ERMT)," (", round(100*mean(Res_Gene.ERMT==1)), "%)",sep=""),
            MLSB=paste(sum(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT!=0)," (", round(100*mean(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT!=0)), "%)",sep=""),
            MEFA=paste(sum(Res_Gene.MEFA), " (", round(100*mean(Res_Gene.MEFA==1)), "%)",sep=""),
            MSRD=paste(sum(Res_Gene.MSRD)," (", round(100*mean(Res_Gene.MSRD==1)), "%)",sep=""),
            M_type=paste(sum(Res_Gene.MEFA+Res_Gene.MSRD!=0), " (", round(100*mean(Res_Gene.MEFA+Res_Gene.MSRD!=0)), "%)",sep=""),
            X23S1_SNP=paste(sum(Res_MUT.X23S1_SNP)," (", round(100*mean(Res_MUT.X23S1_SNP==1)), "%)",sep=""),
            X23S3_SNP=paste(sum(Res_MUT.X23S3_SNP)," (", round(100*mean(Res_MUT.X23S3_SNP==1)), "%)",sep=""),
            Penicillin=paste(sum(Res_MUT.X23S1_SNP+Res_MUT.X23S3_SNP)," (", round(100*mean(Res_MUT.X23S1_SNP+Res_MUT.X23S3_SNP==1)), "%)",sep=""),
            GYRA_SNP=paste(sum(Res_MUT.GYRA_SNP)," (", round(100*mean(Res_MUT.GYRA_SNP==1)), "%)",sep=""),
            PARC_SNP=paste(sum(Res_MUT.PARC_SNP)," (", round(100*mean(Res_MUT.PARC_SNP==1)), "%)",sep=""),
            GYRA_SNP=paste(sum(Res_MUT.GYRA_SNP)," (", round(100*mean(Res_MUT.GYRA_SNP==1)), "%)",sep=""),
            FLuoroquinolones=paste(sum(Res_MUT.PARC_SNP+Res_MUT.GYRA_SNP)," (", round(100*mean(Res_MUT.PARC_SNP+Res_MUT.GYRA_SNP==1)), "%)",sep=""))|> 
  mutate(N= paste("Total (N=", N, ")", sep=""))


# By CC
InSilico_res_per_CC_df <- data_clean|>
  mutate(Res_MUT.X23S1_SNP= ifelse(Res_MUT.X23S1_SNP != "0", 1, 0),
         Res_MUT.X23S3_SNP= ifelse(Res_MUT.X23S3_SNP != "0", 1, 0),
         Res_MUT.GYRA_SNP= ifelse(Res_MUT.GYRA_SNP != "0", 1, 0),
         Res_MUT.PARC_SNP= ifelse(Res_MUT.PARC_SNP != "0", 1, 0)) |>
  group_by(CC)|>
  summarise(N=n(),
            ANT6IA=paste(sum(Res_Gene.ANT6IA), " (", round(100*mean(Res_Gene.ANT6IA==1)), "%)",sep=""),
            APH3III=paste(sum(Res_Gene.APH3III)," (", round(100*mean(Res_Gene.APH3III==1)), "%)",sep=""),,
            Aminoglycosides=paste(sum(Res_Gene.ANT6IA+Res_Gene.APH3III!=0), " (", round(100*mean(Res_Gene.ANT6IA+Res_Gene.APH3III!=0)), "%)",sep=""),
            TETM=paste(sum(Res_Gene.TETM)," (", round(100*mean(Res_Gene.TETM==1)), "%)",sep=""),
            TETO=paste(sum(Res_Gene.TETO)," (", round(100*mean(Res_Gene.TETO==1)), "%)",sep=""),
            Tetracyclines=paste(sum(Res_Gene.TETM+Res_Gene.TETO !=0), " (", round(100*mean(Res_Gene.TETM+Res_Gene.TETO !=0)), "%)",sep=""),
            ERMA=paste(sum(Res_Gene.ERMA)," (", round(100*mean(Res_Gene.ERMA==1)), "%)",sep=""),
            ERMB=paste(sum(Res_Gene.ERMB)," (", round(100*mean(Res_Gene.ERMB==1)), "%)",sep=""),
            ERMT=paste(sum(Res_Gene.ERMT)," (", round(100*mean(Res_Gene.ERMT==1)), "%)",sep=""),
            MLSB=paste(sum(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT!=0)," (", round(100*mean(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT!=0)), "%)",sep=""),
            MEFA=paste(sum(Res_Gene.MEFA), " (", round(100*mean(Res_Gene.MEFA==1)), "%)",sep=""),
            MSRD=paste(sum(Res_Gene.MSRD)," (", round(100*mean(Res_Gene.MSRD==1)), "%)",sep=""),
            M_type=paste(sum(Res_Gene.MEFA+Res_Gene.MSRD!=0), " (", round(100*mean(Res_Gene.MEFA+Res_Gene.MSRD!=0)), "%)",sep=""),
            X23S1_SNP=paste(sum(Res_MUT.X23S1_SNP)," (", round(100*mean(Res_MUT.X23S1_SNP==1)), "%)",sep=""),
            X23S3_SNP=paste(sum(Res_MUT.X23S3_SNP)," (", round(100*mean(Res_MUT.X23S3_SNP==1)), "%)",sep=""),
            Penicillin=paste(sum(Res_MUT.X23S1_SNP+Res_MUT.X23S3_SNP)," (", round(100*mean(Res_MUT.X23S1_SNP+Res_MUT.X23S3_SNP==1)), "%)",sep=""),
            GYRA_SNP=paste(sum(Res_MUT.GYRA_SNP)," (", round(100*mean(Res_MUT.GYRA_SNP==1)), "%)",sep=""),
            PARC_SNP=paste(sum(Res_MUT.PARC_SNP)," (", round(100*mean(Res_MUT.PARC_SNP==1)), "%)",sep=""),
            GYRA_SNP=paste(sum(Res_MUT.GYRA_SNP)," (", round(100*mean(Res_MUT.GYRA_SNP==1)), "%)",sep=""),
            FLuoroquinolones=paste(sum(Res_MUT.PARC_SNP+Res_MUT.GYRA_SNP)," (", round(100*mean(Res_MUT.PARC_SNP+Res_MUT.GYRA_SNP==1)), "%)",sep=""))|> 
  mutate(CC= paste("CC", CC, " (N=", N, ")", sep=""))|> 
  select(-N)
InSilico_res_per_CC_df_t <- cbind(as.data.frame(t(InSilico_res_per_CC_df)), as.data.frame(t(In_silico_df)))
colnames(InSilico_res_per_CC_df_t) <- InSilico_res_per_CC_df_t[1,]
InSilico_res_per_CC_df_t <- InSilico_res_per_CC_df_t[-1, ]
InSilico_res_per_CC_df_t 


```

# Overlap between phenotype and in-silico resistance
```{r}
data_clean |>
  mutate(Erythromycin= factor(Erythromycin, levels=c("s", "r")))|> 
  group_by(Erythromycin)|>
  summarise(N=n(), 
            MLSB=paste(sum(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT!=0)," (", round(100*mean(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT!=0)), "%)",sep=""),
            M_type=paste(sum(Res_Gene.MEFA+Res_Gene.MSRD!=0), " (", round(100*mean(Res_Gene.MEFA+Res_Gene.MSRD!=0)), "%)",sep=""), 
            Total= paste(sum(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT+Res_Gene.MEFA+Res_Gene.MSRD!=0)," (", round(100*mean(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT+Res_Gene.MEFA+Res_Gene.MSRD!=0)), "%)",sep=""))|>
  mutate(Erythromycin = paste(Erythromycin, " (N=", N, ")", sep=""))|>
  select(-N)


data_clean |>
  mutate(Clindamycin= factor(Clindamycin, levels=c("s", "i","r")))|> 
  group_by(Clindamycin)|>
  summarise(N=n(), 
            MLSB=paste(sum(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT!=0)," (", round(100*mean(Res_Gene.ERMA+Res_Gene.ERMB+Res_Gene.ERMT!=0)), "%)",sep=""))|>
  mutate(Clindamycin = paste(Clindamycin, " (N=", N, ")", sep=""))|>
  select(-N)

```

# FIgure 3. 
# Presence/Absence of resistance genes across year, CCs, Serotypes, and clinical severity (ICU & Meningitis)
```{r}
# Distribution per year
df_year <- data_clean |>
  mutate(year.of.cx = year(Date.of.cx)) |> 
  group_by(year.of.cx) |> 
  summarise(ANT6IA = mean(Res_Gene.ANT6IA == 1),
            APH3III = mean(Res_Gene.APH3III == 1),
            TETM = mean(Res_Gene.TETM == 1),
            TETO = mean(Res_Gene.TETO == 1),
            ERMA = mean(Res_Gene.ERMA == 1),
            ERMB = mean(Res_Gene.ERMB == 1),
            ERMT = mean(Res_Gene.ERMT == 1),
            MEFA = mean(Res_Gene.MEFA == 1),
            MSRD = mean(Res_Gene.MSRD == 1)) |> 
  pivot_longer(cols = -year.of.cx,  # Pivot all columns except 'year.of.cx'
               names_to = "Gene",   # Name the gene column
               values_to = "Proportion")
heatmap_year <- ggplot(df_year, aes(x = Gene, y = as.factor(year.of.cx), fill = Proportion)) +
  geom_tile() +
  scale_fill_gradient(low = "white", high = "blue") +  # Customize colors
  labs(title = "Proportion of Resistance Genes by Year", y = "Year", x = "Resistance Gene", fill = "Proportion") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels for readability

# Distribution per CC
df_CC <- data_clean |>
  group_by(CC) |> 
  summarise(ANT6IA = mean(Res_Gene.ANT6IA == 1),
            APH3III = mean(Res_Gene.APH3III == 1),
            TETM = mean(Res_Gene.TETM == 1),
            TETO = mean(Res_Gene.TETO == 1),
            ERMA = mean(Res_Gene.ERMA == 1),
            ERMB = mean(Res_Gene.ERMB == 1),
            ERMT = mean(Res_Gene.ERMT == 1),
            MEFA = mean(Res_Gene.MEFA == 1),
            MSRD = mean(Res_Gene.MSRD == 1)) |> 
  pivot_longer(cols = -CC,  
               names_to = "Gene",   
               values_to = "Proportion")
heatmap_CC <- ggplot(df_CC, aes(x = Gene, y = as.factor(CC), fill = Proportion)) +
  geom_tile() +
  scale_fill_gradient(low = "white", high = "orange") + 
  labs(title = "Proportion of Resistance Genes by CC", y = "CC", x = "Resistance Gene", fill = "Proportion") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  


# Distribution per Serotype
df_Serotype <- data_clean |>
  group_by(Serotype) |> 
  summarise(ANT6IA = mean(Res_Gene.ANT6IA == 1),
            APH3III = mean(Res_Gene.APH3III == 1),
            TETM = mean(Res_Gene.TETM == 1),
            TETO = mean(Res_Gene.TETO == 1),
            ERMA = mean(Res_Gene.ERMA == 1),
            ERMB = mean(Res_Gene.ERMB == 1),
            ERMT = mean(Res_Gene.ERMT == 1),
            MEFA = mean(Res_Gene.MEFA == 1),
            MSRD = mean(Res_Gene.MSRD == 1)) |> 
  pivot_longer(cols = -Serotype,  
               names_to = "Gene",   
               values_to = "Proportion")
heatmap_Serotype <- ggplot(df_Serotype, aes(x = Gene, y = as.factor(Serotype), fill = Proportion)) +
  geom_tile() +
  scale_fill_gradient(low = "white", high = "yellow3") + 
  labs(title = "Proportion of Resistance Genes by Serotype", y = "Serotype", x = "Resistance Gene", fill = "Proportion") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  


# Distribution per ICU
df_ICU <- data_clean |>
  group_by(ICU) |> 
  summarise(ANT6IA = mean(Res_Gene.ANT6IA == 1),
            APH3III = mean(Res_Gene.APH3III == 1),
            TETM = mean(Res_Gene.TETM == 1),
            TETO = mean(Res_Gene.TETO == 1),
            ERMA = mean(Res_Gene.ERMA == 1),
            ERMB = mean(Res_Gene.ERMB == 1),
            ERMT = mean(Res_Gene.ERMT == 1),
            MEFA = mean(Res_Gene.MEFA == 1),
            MSRD = mean(Res_Gene.MSRD == 1)) |> 
  pivot_longer(cols = -ICU,  
               names_to = "Gene",   
               values_to = "Proportion")
heatmap_ICU <- ggplot(df_ICU, aes(x = Gene, y = as.factor(ICU), fill = Proportion)) +
  geom_tile() +
  scale_fill_gradient(low = "white", high = "darkred") + 
  labs(title = "Proportion of Resistance Genes by ICU", y = "ICU", x = "Resistance Gene", fill = "Proportion") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) 


# Distribution per Meningitis
df_Meningitis <- data_clean |>
  group_by(Meningitis) |> 
  summarise(ANT6IA = mean(Res_Gene.ANT6IA == 1),
            APH3III = mean(Res_Gene.APH3III == 1),
            TETM = mean(Res_Gene.TETM == 1),
            TETO = mean(Res_Gene.TETO == 1),
            ERMA = mean(Res_Gene.ERMA == 1),
            ERMB = mean(Res_Gene.ERMB == 1),
            ERMT = mean(Res_Gene.ERMT == 1),
            MEFA = mean(Res_Gene.MEFA == 1),
            MSRD = mean(Res_Gene.MSRD == 1)) |> 
  pivot_longer(cols = -Meningitis,  
               names_to = "Gene",   
               values_to = "Proportion")
heatmap_Meningitis <- ggplot(df_Meningitis, aes(x = Gene, y = as.factor(Meningitis), fill = Proportion)) +
  geom_tile() +
  scale_fill_gradient(low = "white", high = "darkred") + 
  labs(title = "Proportion of Resistance Genes by Meningitis", y = "Meningitis", x = "Resistance Gene", fill = "Proportion") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) 


# Join
library(patchwork)
heatmap <- (heatmap_year) |(heatmap_Serotype /  heatmap_CC) | (heatmap_ICU / heatmap_Meningitis ) 
heatmap
```


# Identifying correlates  of severe clinical outcomes
# TABLE 2   
```{r}
# Identify potential confounders: 
# data_clean_ICU_OR <- data_clean |> filter(days_at_dx <= 365)
# ## Age Categorical
# chisq.test(data_clean_ICU_OR $ICU, data_clean_ICU_OR$Age_cat)
# ## Age Continuous (Days at Dx)
# ### Test Association between Age and Outcome
# wilcox.test(days_at_dx ~ Admission_loc, data = data_clean_ICU_OR)
# Age_ICU_mod <- glm(ICU ~ days_at_dx, data = data_clean_ICU_OR, family = binomial)
# summary(Age_ICU_mod)
# ### check distribution
# data_clean_ICU_OR |> 
#   ggplot()+
#   geom_histogram(aes(x=days_at_dx))+
#   facet_grid(ICU ~.)
# ### Show median + IQR
# data_clean_ICU_OR |> 
#   group_by(Admission_loc)|> 
#   summarise(median_age= paste0(median(days_at_dx), " (IQR:",quantile(days_at_dx, 0.05), ";", quantile(days_at_dx, 0.95), ")"))
# ### Test Association between Age and Exposures
# Age_ICU_mod <- glm(ICU ~ days_at_dx, data = data_clean_ICU_OR, family = binomial)
# get_OR <- function(risk_factor) {
#   risk_factor_col <- data_clean_ICU_OR[[risk_factor]]
#   coef(summary(glm(ICU ~ risk_factor_col, data = data_clean_ICU_OR, family = binomial)))[2,4] 
# }
# 
# risk_factors <- colnames(data_clean_ICU_OR |> select(contains(c("Surface_protein.", "Virulence_Gene."))))
# 
# results <- lapply(risk_factors, get_OR)
# get_OR(risk_factors[1])
# get_OR(risk_factors[2])
# get_OR(risk_factors[3])
# get_OR(risk_factors[4])
# get_OR(risk_factors[5])
# get_OR(risk_factors[6])
# get_OR(risk_factors[7])
# get_OR(risk_factors[8])
# get_OR(risk_factors[9])
# get_OR(risk_factors[10])
# get_OR(risk_factors[11])
# get_OR(risk_factors[12])
# get_OR(risk_factors[13])
# 
# ## CC
# chisq.test(data_clean_ICU_OR $ICU, data_clean_ICU_OR$CC)
# ## ST
# chisq.test(data_clean_ICU_OR $ICU, data_clean_ICU_OR$ST)
# ## Serotype
# chisq.test(data_clean_ICU_OR $ICU, data_clean_ICU_OR$Serotype)
# summary(glm(ICU ~ Serotype, data = data_clean_ICU_OR, family = binomial))
# ## Meningitis
# chisq.test(data_clean_ICU_OR $ICU, data_clean_ICU_OR$Meningitis)
# Meningitis_ICU_mod <- glm(ICU ~ Meningitis, data = data_clean_ICU_OR, family = binomial)
# summary(Meningitis_ICU_mod)


# Function 
calculate_or <- function(df, risk_factor) {
  
  # Ensure that risk_factor is treated as a column from the data_cleanframe
  risk_factor_col <- df[[risk_factor]]
  
  # Create a 2x2 table of ICU vs. Other by the risk factor
  ICU <- sum(df |> filter(ICU == 1) |> pull(risk_factor))
  Other <- sum(df |> filter(ICU == 0) |> pull(risk_factor)) 
  
  # Logistic Regression
  log_reg <- glm(ICU ~ risk_factor_col, data = df, family = binomial)
  log_reg_summary <- coef(summary(log_reg))
  
  # Extract Odds Ratio (exponentiate the coefficient) and Confidence Intervals
  or_value <- exp(log_reg_summary[2, 1])
  ci_lower <- exp(log_reg_summary[2, 1] - 1.96 * log_reg_summary[2, 2])
  ci_upper <- exp(log_reg_summary[2, 1] + 1.96 * log_reg_summary[2, 2])
  
  # Extract P-value
  p_value <- round(log_reg_summary[2, 4],3)
  
  # Return the formatted OR, CI, and P-value
  data.frame(
    Risk_Factor = risk_factor,
    "ICU (n=31)" = ICU,
    "Other (n=39)" = Other,
    "OR (95% CI)" = sprintf("%.2f (%.2f - %.2f)", or_value, ci_lower, ci_upper),
    "P-value" = p_value
  )
}

# List of risk factors to test
risk_factors <- colnames(data_clean |> 
                           select(contains(c("Surface_protein.", "Virulence_Gene.")))|> 
                           select(-Surface_protein.Sip))


# Calculate OR, CI, and P-value for each risk factor
results <- lapply(risk_factors, calculate_or, df = data_clean_ICU_OR)

# Combine the results into a single data_clean frame
final_table <- bind_rows(results)

final_table |> 
  mutate(Risk_Factor = str_remove(str_remove(Risk_Factor, "Surface_protein."), "Virulence_Gene."))

```